ansibullbot/triagers/plugins/docs_info.py

```diff
@@ -0,0 +1,306 @@
+            if source.doc_string:
+                if source.ds_line_start <= line["lineno"] <= source.ds_line_end:
+                    continue
+
+            # Check if change applies to module EXAMPLES (if exists)
+            if source.example_string:
+                if source.ex_line_start <= line["lineno"] <= source.ex_line_end:
+                    continue
+
+            # Find appropriate class if it exists
+            source_class = source.find_class(line["lineno"])
+            if source_class is not None:
+                if source_class.doc_string:
+                    if source_class.ds_line_start <= line["lineno"] <= source_class.ds_line_end:
+                        continue
+
+                source_func = source_class.find_function(line["lineno"])
+                if source_func is not None and source_func.doc_string:
+                    if source_func.ds_line_start <= line["lineno"] <= source_func.ds_line_end:
+                        continue
+
+            # If we made it this far, this change is outside docs/examples
+            return False
+        else:
+            return True
+
+def get_docs_facts(iw):
+    """ Cycle through the files and triage if changes only
+        affect docs-related files/content.
+    """
+    dfacts = {
+        "is_docs_only": False
+    }
+
+    if not iw.is_pullrequest():
+        return dfacts
+
+    docs_only = True
+
+    for commit in iw.commits:
+        commit_files = iw.get_commit_files(commit)
+        if commit_files is None:
+            # "Sorry, this diff is temporarily unavailable due to heavy server load."
+            # Preserve docsite label to prevent potential waffling
+            dfacts["is_docs_only"] = "docsite" in iw.labels
+            return dfacts
+
+        for changed_file in commit_files:
+            if isinstance(changed_file, dict):
+                changed_file = CommitFile(changed_file)
+
+            if _is_docs_path(changed_file.filename):
+                continue
+            else:
+                # Additions or deletions of complete files outside of
+                # docsite/examples will likely be more than documentation
+                # changes
+                if changed_file.status != "modified":
+                    docs_only = False
+                    break
+                else:
+                    # Non-Python files that are outside of the docsite
+                    # folder will not be documentation related
+                    if not changed_file.filename.endswith(".py"):
+                        docs_only = False
+                        break
+                    # If a python file, check if the changes are only to
+                    # docstrings
+                    else:
+                        docs_only = _check_py_changes(
+                                changed_file.file_content,
+                                changed_file.patch
+                        )
+
+        if not docs_only:
```

Comment:
Seems like this is incorrectly indented as it should be checked per `changed_file`.

Reply:
This is to break out of the parent loop that is iterating over each commit. This is basically implementing a "fail fast" scenario, since there was a non-docs change and no need to evaluate the rest.

However, this comment gives me the thought that iterating over the commits is extraneous. I used the `small_patch.py` plugin as a template to munge, which lead me to iterate over commits. My original prototype simply iterated over the file 
info returned via a base...head comparison (`pygithub.repo.compare()`). I could ostensibly use `issuewrapper.get_files()` to get only the file info we use here.

Reply:
> This is to break out of the parent loop that is iterating over each commit. This is basically implementing a "fail fast" scenario, since there was a non-docs change and no need to evaluate the rest.

I guess my point was that _check_py_changes is called per changed_file but nothing is done with the result. Shouldn't we break out of the "per changed_file" loop as soon as _check_py_changes returns False?

> However, this comment gives me the thought that iterating over the commits is extraneous. I used the small_patch.py plugin as a template to munge, which lead me to iterate over commits. My original prototype simply iterated over the file
info returned via a base...head comparison (pygithub.repo.compare()). I could ostensibly use issuewrapper.get_files() to get only the file info we use here.

+1

Reply:
Using `issuewrappper.get_files()` (broken, btw) and/or `issuewrapper.files()` doesn't work how I assumed. They only return the filename and the sha, from the meta info. I'd likely have to build out a new solution to get something like `compare base...head`, for the relevant API.

Attempting this though did highlight that the purpose of iterating over commits may be so that the meta history can be built.

I'll stick with iterating over the commits for now, but will work to get a better understanding of how that affects the desired outcome.